docs/internals/optimizer.rst

```diff
@@ -933,7 +934,7 @@ DeadCodeEliminator
 This optimization stage removes unreachable code.
 
 Unreachable code is any code within a block which is preceded by a
-leave, return, invalid, break, continue, selfdestruct or revert.
+leave, return, invalid, break, continue, selfdestruct or revert (whether it be an explicit ``revert()``, or caused by infinite recursion).
```

Comment:
Note that ``invalid()`` earlier in the list also is a case of a revert (we may also want to add that to the UnusedStoreEliminator docs)

Reply:
That's why I was more for the option of just using revert as a blanket statement, instead of mentioning specific reasons that a revert may occur. Aside from `invalid()` (which is explicit I guess?), is there anything else that may cause a revert?

Reply:
You can reverse the point, though - since here you have an explicit list of statements and builtins causing reverts, "revert" specifically *doesn't* cover ``invalid`` nor infinite recursion. If ``revert`` was meant to cover it, ``invalid`` wouldn't be in the list.
But no, there's no other cases. Well, theoretically, infinite loops, but the optimizer doesn't handle them specially.

Reply:
Yes, I understand the point, and I do somewhat agree with it, but my understanding is that we have a blanket concept of a revert, which may be caused by explicit calls, such as `revert()` and `invalid()`, and by ones generated implicitly through an optimizer step, such as the infinite recursion case. The unfortunate thing here I guess, is the fact that a `revert()` is a subset of revert, so a `revert()` is a revert, but a revert isn't a `revert()` only :D 
And I guess it's a little late to introduce a term like rollback.